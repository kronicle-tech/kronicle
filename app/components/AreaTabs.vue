<template>
  <div class="my-3">
    <b-button class="my-1" :to="`/areas/${areaId}`"> Info </b-button>
    <b-button class="my-1" :to="`/areas/${areaId}/dependencies`">
      Visualizations
    </b-button>
    <b-button class="my-1" :to="`/areas/${areaId}/components`">
      Components
    </b-button>
    <b-button
      class="my-1"
      :to="`/areas/${areaId}/cross-functional-requirements`"
    >
      Cross Functional Requirements
    </b-button>
    <b-button class="my-1" :to="`/areas/${areaId}/tech-debts`">
      Tech Debts
    </b-button>
    <b-button class="my-1" :to="`/areas/${areaId}/tests`"> Tests </b-button>
    <b-button class="my-1" :to="`/areas/${areaId}/errors`"> Errors </b-button>
    <b-button class="my-1" :to="`/areas/${areaId}/openapi-specs`">
      OpenAPI Specs
    </b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
  props: {
    areaId: {
      type: String,
      required: true,
    },
  },
})
</script>
