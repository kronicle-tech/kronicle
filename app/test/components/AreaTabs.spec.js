import { mount } from '@vue/test-utils'
import AreaTabs from '@/components/AreaTabs.vue'

describe('AreaTabs', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(AreaTabs, { propsData })
  }

  beforeEach(() => {
    propsData = {
      areaId: 'test-area-id',
    }
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  test('renders the component', () => {
    createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
