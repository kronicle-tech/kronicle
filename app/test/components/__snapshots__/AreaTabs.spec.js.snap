// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaTabs renders the component 1`] = `
"<div class=\\"my-3\\"><a href=\\"/areas/test-area-id\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\"> Info </a> <a href=\\"/areas/test-area-id/dependencies\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Visualizations
  </a> <a href=\\"/areas/test-area-id/components\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Components
  </a> <a href=\\"/areas/test-area-id/cross-functional-requirements\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Cross Functional Requirements
  </a> <a href=\\"/areas/test-area-id/tech-debts\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Tech Debts
  </a> <a href=\\"/areas/test-area-id/tests\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\"> Tests </a> <a href=\\"/areas/test-area-id/errors\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\"> Errors </a> <a href=\\"/areas/test-area-id/openapi-specs\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    OpenAPI Specs
  </a></div>"
`;
